subprocess/cmd.go

```diff
@@ -0,0 +1,110 @@
+package subprocess
+
+import (
+	"io"
+	"os"
+	"os/exec"
+)
+
+// Thin wrapper around exec.Cmd. Takes care of pipe shutdown by
+// keeping an internal reference to any created pipes. Whenever
+// Cmd.Wait() is called, all created pipes are closed.
+type SafeCmd struct {
+	Path string
+	Args []string
+	Env  []string
+	Dir  string
+
+	Stdin  io.Reader
+	Stdout io.Writer
+	Stderr io.Writer
+
+	Process *os.Process
+
+	cmd        *exec.Cmd
```

Comment:
I actually tried this the first time since I'd read about Go's inheritance-like ability, but I got some pointer errors because I didn't realize you had to initialize the child struct like this. Thanks for the tip, I got it to work!